src/ralph/models/edx/textbook_interaction.py

```diff
@@ -0,0 +1,627 @@
+"""Textbook interaction event model definitions"""
+
+from typing import Literal, Optional, Union
+
+from pydantic import Field, Json, constr
+
+from ralph.models.selector import selector
+
+from .base import AbstractBaseEventField
+from .browser import BaseBrowserEvent
+
+
+class TextbookInteractionBaseEventField(AbstractBaseEventField):
+    """Represents the event field which attributes are common to most of the textbook
+    interactions events.
+
+    Attributes:
+        chapter (str): Consists of the name of the PDF file.
+            It begins with the `block_id` value and ends with the `.pdf` extension.
+        page (int): The number of the page that is open when the event is emitted.
+    """
+
+    page: int
+    chapter: constr(
+        regex=(
+            r"^block-v1:[^\/+]+(\/|\+)[^\/+]+(\/|\+)[^\/?]+type"  # noqa : F722
+            r"@sequential\+block@[a-f0-9]{32}.+\.pdf$"  # noqa : F722
+        )
+    )
+
+
+class TextbookPdfThumbnailsToggledEventField(TextbookInteractionBaseEventField):
+    """Representents the `textbook.pdf.thumbnails.toggled` event field.
```

Comment:
```suggestion
    """Represents the `textbook.pdf.thumbnails.toggled` event field.
```
(This seems to be in multiple places)
s/Representents/Represents/